backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -435,4 +442,92 @@ abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHit
 
   def sourcePathFromCopyOutputsCommand(command: CopyOutputsCommand): String = command.jobDetritusFiles.values.head
 
-}
+  private def issueSourceLocationCommand(copyCommand: CopyOutputsCommand): State = {
+    val sourcePath = sourcePathFromCopyOutputsCommand(copyCommand)
+    log.warning(s"Willy, in issueSourceLocationCommand, copyCommand.source is $sourcePath")
+    commandBuilder.locationCommand(getPath(sourcePath).get) match {
+      case Success(command) =>
+        sendIoCommand(command)
+        goto(CheckingCacheHitLocation)using
+          Option(SourceLocationCheckData(copyCommand))
+      case Failure(e) =>
+        failAndStop(CopyAttemptError(new Throwable(s"Error building location command for file path '$sourcePath''", e)))
+    }
+  }
+
+  private def issueDestinationLocationCommand(sourceLocation: String, copyCommand: CopyOutputsCommand): State = {
```

Comment:
Thanks for the review, Miguel! Do you have a suggestion on how best to achieve this? Its not clear to me how to avoid doing this for every job.

Reply:
After discussing with Miguel today we've agreed to leave the destination location checking as-is (at the job level). A few reasons for this are:
1) At the workflow level, we don't know the backend type.
2) We could cache locations per bucket at the job level, but it would add unnecessary complication to add a caching layer across all JobActors.